src/main/java/com/hivemq/adapter/sdk/api/tag/Tag.java

```diff
@@ -17,13 +17,15 @@
 
 import org.jetbrains.annotations.NotNull;
 
-public interface Tag<T> {
+public interface Tag {
 
     @NotNull
-    T getTagDefinition();
+    TagDefinition getDefinition();
```

Comment:
Would T extends TagDefinition make sense here?

Reply:
Nope. The JVM and its shitty implementation of generic causes issues.
There are multiple places where because of serialization and guessing we end up without any typeinfo which makes the code even worse.
Maybe we have a look together?